apis/v1alpha1/validation.go

```diff
@@ -33,11 +37,64 @@ func ValidateResourceMapping(m *ResourceMapping) (vErr error) {
 			vErr = errors.Join(vErr, fmt.Errorf("invalid owner: %w", err))
 		}
 	}
-	if m.Resource.Provider == "" {
-		vErr = errors.Join(vErr, fmt.Errorf("empty resource provider"))
-	}
+
 	if _, ok := m.Annotations.AsMap()[AnnotationKeyAssetInfo]; ok {
 		vErr = errors.Join(vErr, fmt.Errorf("reserved key is included: %s", AnnotationKeyAssetInfo))
 	}
+
+	if err := validateResource(m.Resource); err != nil {
+		vErr = errors.Join(vErr, fmt.Errorf("invalid Resourc: %w", err))
+	}
+
 	return
 }
+
+func validateResource(r *Resource) (vErr error) {
+	if r.Name == "" {
+		vErr = errors.Join(vErr, fmt.Errorf("empty resource name"))
+	}
+
+	if r.Provider == "" {
+		vErr = errors.Join(vErr, fmt.Errorf("empty resource provider"))
+	}
+
+	if err := validateAndNormalizeSubscope(r); err != nil {
+		vErr = errors.Join(vErr, fmt.Errorf("invalid subscope: %w", err))
+	}
+
+	return
+}
+
+func validateAndNormalizeSubscope(r *Resource) error {
+	if r.Subscope == "" {
+		return nil
+	}
+
+	// normalize subscope string to only have lower cases
```

Comment:
This is not normalize, this is just a validation check. In addition, I think we can't enforce value to be lower case, you can just enforce key? for example, gcs object can include upper case.

Reply:
Update to only check is keys are in lower case.

Reply:
No longer check if keys are in lower case.